Mathlib/GroupTheory/Subgroup/Centralizer.lean

```diff
@@ -94,4 +94,22 @@ abbrev closureCommGroupOfComm {k : Set G} (hcomm : ∀ x ∈ k, ∀ y ∈ k, x *
       have := closure_le_centralizer_centralizer k
       Subtype.ext <| Set.centralizer_centralizer_comm_of_comm hcomm _ (this h₁) _ (this h₂) }
 
+/-- The homomorphism N(H) → Aut(H) with kernel C(H). -/
+@[simps]
+def normalizerMonoidHom : H.normalizer →* MulAut H where
```

Comment:
Do we have a `MulAction H.normalizer H` with `MulDistribMulAction` instance? If yes, then this will follow from the general theory.

Reply:
I suggest that you add instances for `MulAction H.normalizer H` and `MulDistribMulAction H.normalizer H` (by moving parts of your proof to the instances), then use [MulDistribMulAction.toMulAut](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/GroupWithZero/Action/Basic.html#MulDistribMulAction.toMulAut) to get the hom. This way `SMul` doesn't depend on bundled homomorphisms, which may be good for resolving defeq diamonds in the future.